translator/translate/otel/pipeline/prometheus/translator.go

```diff
@@ -12,41 +13,101 @@ import (
-	return component.NewIDWithName(component.DataTypeMetrics, pipelineName)
+	return component.NewIDWithName(component.DataTypeMetrics, t.name)
 }
 
 // Translate creates a pipeline for prometheus if the logs.metrics_collected.prometheus
 // section is present.
 func (t *translator) Translate(conf *confmap.Conf) (*common.ComponentTranslators, error) {
-	key := common.ConfigKey(common.LogsKey, common.MetricsCollectedKey, common.PrometheusKey)
-	if conf == nil || !conf.IsSet(key) {
-		return nil, &common.MissingKeyError{ID: t.ID(), JsonKey: key}
+	if conf == nil || !(conf.IsSet(MetricsKey) || conf.IsSet(LogsKey)) {
+		return nil, &common.MissingKeyError{ID: t.ID(), JsonKey: fmt.Sprint(MetricsKey + " or " + LogsKey)}
+	}
+
+	// return pipeline based on destination to keep source/destination combinations clearly separated
+	// telegraf_prometheus - cloudwatch
+	// otel_prometheus - AMP
+	// this could change in future releases to support different source/destination combinations
+	var destinations []string
+	if t.configSection == LogsKey {
+		destinations = append(destinations, common.GetLogsDestinations()...)
+	} else if t.configSection == MetricsKey {
+		destinations = append(destinations, common.GetMetricsDestinations(conf)...)
+	}
+	// each matching case returns 1 component translator
+	// but this could also follow the translators pattern that handles destinations then merge all returned translators
+	for _, destination := range destinations {
+		switch destination {
+		case common.CloudWatchLogsKey:
+			if !conf.IsSet(LogsKey) {
+				return nil, fmt.Errorf("pipeline (%s) is missing prometheus configuration under logs section with destination (%s)", t.name, destination)
+			}
+			return &common.ComponentTranslators{
+				Receivers: common.NewTranslatorMap(adapter.NewTranslator(prometheus.SectionKey, LogsKey, time.Minute)),
+				Processors: common.NewTranslatorMap(
+					batchprocessor.NewTranslatorWithNameAndSection(t.name, common.LogsKey), // prometheus sits under metrics_collected in "logs"
+				),
+				Exporters:  common.NewTranslatorMap(awsemf.NewTranslatorWithName(common.PipelineNamePrometheus)),
+				Extensions: common.NewTranslatorMap(agenthealth.NewTranslator(component.DataTypeLogs, []string{agenthealth.OperationPutLogEvents})),
+			}, nil
+		case common.AMPKey:
+			if !conf.IsSet(MetricsKey) {
+				return nil, fmt.Errorf("pipeline (%s) is missing prometheus configuration under metrics section with destination (%s)", t.name, destination)
+			}
+			if !conf.IsSet(common.ConfigKey(MetricsKey, common.PrometheusConfigPathKey)) {
+				return nil, &common.MissingKeyError{ID: t.ID(), JsonKey: fmt.Sprint(common.ConfigKey(MetricsKey, common.PrometheusConfigPathKey))}
+			}
+			if !conf.IsSet(AMPSectionKey) || !conf.IsSet(common.ConfigKey(AMPSectionKey, common.WorkspaceIDKey)) {
+				return nil, &common.MissingKeyError{ID: t.ID(), JsonKey: fmt.Sprint(AMPSectionKey + " or " + common.ConfigKey(AMPSectionKey, common.WorkspaceIDKey))}
+			}
```

Comment:
Don't include this check. The `prometheusremotewrite` translator already has it and will fail the translation, which is what we want if the AMP section is not configured and the `metrics/metrics_collected/prometheus` section is.